### PR TITLE
[IR-754] Iroha Special Instructions related to Iroha events.

### PR DESCRIPTION
**Pull request by humb1t**

### CODE REVIEW

File: iroha/docs/references/glossary.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+### Out of the Box Iroha Special Instruction
+
+Out of the Box Iroha provides several basic Instructions for utility purposes or domains
+related functionality.
+
+#### Utilitary Iroha Special Instruction
+
+This set contains logical instructions like `If`, I/O related like `Notify` and compositions
+like `Sequence`. They mostly used by [custom Instructions](#custom-iroha-special-instruction).
```
Comment (eadventurous) [inline iroha/docs/references/glossary.md:25]: `They are mostly used by [Custom Instructions](...`

File: iroha/docs/references/glossary.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+#### Domains related Iroha Special Instruction
+
+This set contains domains related instructions (asset/account/domain/peer) and provides
+an ability to make changes on the [World State View](#world-state-view) in a safe way.
```
Comment (eadventurous) [inline iroha/docs/references/glossary.md:30]: `make changes to the ...`

File: iroha/docs/references/glossary.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+### Custom Iroha Special Instruction
+
+These Instructions provided by [Iroha Modules](#todo), clients or 3rd parties. They can be only
```
Comment (eadventurous) [inline iroha/docs/references/glossary.md:34]: `These instructions are provided...`

File: iroha/src/event.rs [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+    pub enum EventInstruction {
+        /// This variant of Iroha Special Instruction will execute instruction when new Block
+        /// will be created.
+        OnBlockCreated(Box<Instruction>),
+        /// This variant of Iroha Special Instruction will execute instruction when Blockchain
+        /// will reach predefined height.
+        OnBlockchainHeight(u64, Box<Instruction>),
+        /// This variant of Iroha Special Instruction will execute instruction when World State
+        /// View change will be detected.
+        OnWorldStateViewChange(Box<Instruction>, Box<Instruction>),
```
Comment (eadventurous) [inline iroha/src/event.rs:27]: The comment does not explain that the first parameter is a trigger and only the second is an instruction. I think it should also point it out and maybe add `Trigger` type alias.

File: iroha/src/event.rs [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+                        .height
+                        == height
+                    {
+                        instruction.execute(authority, world_state_view)
+                    } else {
+                        Ok(())
+                    }
+                }
+                OnWorldStateViewChange(trigger, instruction) => {
+                    if trigger.execute(authority.clone(), world_state_view).is_ok() {
```
Comment (eadventurous) [inline iroha/src/event.rs:57]: To my mind a trigger should not change state, but should be similar to the `Query` but with boolean value.

Comment (humb1t) [reply iroha/src/event.rs:57]: https://jira.hyperledger.org/browse/IR-754?focusedCommentId=69460&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-69460

**Pull request merged**